pyro/infer/mcmc/api.py

```diff
@@ -377,6 +374,12 @@ def model(data):
         z_flat_acc = [[] for _ in range(self.num_chains)]
         with optional(pyro.validation_enabled(not self.disable_validation),
                       self.disable_validation is not None):
+            if self.num_chains > 1:
+                # XXX we clone CUDA tensor args to resolve the issue "Invalid device pointer"
+                # at https://github.com/pytorch/pytorch/issues/10375
+                # This also resolves "RuntimeError: Cowardly refusing to serialize non-leaf tensor which
+                # requires_grad", which happens with `jit_compile` under PyTorch 1.7
+                args = [arg.clone().detach() if torch.is_tensor(arg) else arg for arg in args]
```

Comment:
LGTM. I'm curious - any idea why this is happening now and do we have a smaller example that reproduces this?